cli/js/lib.deno.shared_globals.d.ts

```diff
@@ -606,12 +606,12 @@ type FormDataEntryValue = File | string;
  * XMLHttpRequest.send() method. It uses the same format a form would use if the
  * encoding type were set to "multipart/form-data". */
 interface FormData extends DomIterable<string, FormDataEntryValue> {
-  append(name: string, value: string | Blob, fileName?: string): void;
+  append(name: string, value: string | Blob | File, fileName?: string): void;
```

Comment:
This type doesn't appear in in [lib.dom.d.ts](https://github.com/microsoft/TypeScript/blob/7fc456f2d70348725bbef92732b98a33c07f85ad/lib/lib.dom.d.ts#L5549)
```ts
/** Provides a way to easily construct a set of key/value pairs representing form fields and their values, which can then be easily sent using the XMLHttpRequest.send() method. It uses the same format a form would use if the encoding type were set to "multipart/form-data". */
interface FormData {
    append(name: string, value: string | Blob, fileName?: string): void;
    delete(name: string): void;
    get(name: string): FormDataEntryValue | null;
    getAll(name: string): FormDataEntryValue[];
    has(name: string): boolean;
    set(name: string, value: string | Blob, fileName?: string): void;
    forEach(callbackfn: (value: FormDataEntryValue, key: string, parent: FormData) => void, thisArg?: any): void;
}
```

Reply:
hmm that is interesting... because a File is valid input for a FormData. Tho since File extends on Blob, that might be the reason?

Reply:
I just checked the spec, and it does allow a File, but the interface only defines the type of Blob as passable argument. I guess they rely on the fact that File extends on Blob.

Reply:
So either we remove File from https://github.com/denoland/deno/blob/76a6a1ff46adde01877fbf4b874dc8d344a74f9c/cli/js/web/form_data.ts#L15-L19 (and same for the `set` method) but still allow a File, or we accept this PR.

Reply:
@crowlKats I think it would be best if we match the TS declarations exactly - they're probably correct. So, I'm in favor of removing File from cli/js/web/form_data.ts as you suggested...

Reply:
hmm i see, tho all browsers do support it, and the [spec does somewhat mention it implicitly](https://xhr.spec.whatwg.org/#concept-formdata-entry). [MDN also mentions it more explicitly (under the `value` param)](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append#Syntax). I wouldnt mind removing said feature from `form_data.ts`, but it does improve experience, and i aint sure if we want to go against other browsers...

Reply:
@ry (sorry, i forgot once again to tag you)

Reply:
We should match the `dom` because it is based off the WebIDL.  `File` should be assignable to `Blob`, and so there is no need for extra public type.  Because `File` extends `Blob` that is why the documentation isn't explicit about it.  The internal code shouldn't matter, but we "should" be able to eliminate `domFile.DomFileImpl` from the code.  That doesn't mean we don't support it, it just means that all dom `Blob`s are supported, which dom `File` is a `Blob`.

Reply:
@kitsonk alright, could you take a look at the latest changes?